### ISSUE TITLE
Gist just saying You dont have configured github account

### ISSUE THREAD

**Issue by favetelinguis** [mattn/gist-vim#199]
What can I do to find the error when gist only says You dont have github account configured?

I have done the follwoing:
Install web-api and gist with vundle
Run git --global github.user=... and git --gloab github.token=...
And created .gist-vim and put token=... but still gist just tells me I dont have github accound configured.

I get the same error when trying to run gist -a

Not sure what more i can try...

**Comment by mattn** [comment]
try to remove `~/.gist-vim`.

**Comment by favetelinguis** [comment]
Makes no difference.
Den 9 nov 2015 01:15 skrev "mattn" <notifications@github.com>:

**Comment by mattn** [comment]
What message do you get from below?

```
:echo g:github_user
```

What response do you get from below?

```
:echo substitute(s:system('git config --get github.user'), "\n", '', '')
```

**Comment by favetelinguis** [comment]
First one: nothing is returned

Secound one:
E15: Invalid expression: substitute(s:system('git config --get github.user'), "\n", '','')        
E120: Using <SID> not in a script context: s:system                                               
E116: Invalid arguments for function substitute(s:system('git config --get github.user'), "\n", ''
,'')

**Comment by mattn** [comment]
Ah, sorry. try this instead.

```
:echo substitute(system('git config --get github.user'), "\n", '','')
```

**Comment by favetelinguis** [comment]
That did show my github account name!?!

**Comment by mattn** [comment]
Yes, you will see your account name because it's given from https://github.com/mattn/gist-vim/blob/master/autoload/gist.vim#L39

So the error https://github.com/mattn/gist-vim/blob/master/autoload/gist.vim#L719 couldn't be displayed.

**Comment by favetelinguis** [comment]
What might be the reason im not seeing the account name in the firs example?
Den 11 nov 2015 00:52 skrev "mattn" <notifications@github.com>:

**Comment by mattn** [comment]
> What might be the reason im not seeing the account name in the firs example?

Removed by another plugin.

**Comment by mahloun** [comment]
I've got the same error here and some things don't appear clear to me:

@mattn you asked @favetelinguis to perform in ex mode, a function's call that we can find in line #39 of the [script](https://github.com/mattn/gist-vim/blob/master/autoload/gist.vim#L39):
`substitute(s:system('git config --get github.user'), "\n", '', '')`
which gave an error (like for me). You put on the fault of the `s:` and, thus you asked to remove it and run again, which resulted in the following function's call that was just fine (like for me):
`substitute(system('git congit --get github.user'), "\n", '', '')`.

My question is why keeping the failed function's call in the source code instead of replacing it by the working one?

Also @mattn I really didn't understand this reply of yours 

> Yes, you will see your account name because it's given from https://github.com/mattn/gist-vim/blob/master/autoload/gist.vim#L39
So the error https://github.com/mattn/gist-vim/blob/master/autoload/gist.vim#L719 couldn't be displayed.

**Comment by mattn** [comment]
What error do you get from `git config --get github.user` ?

**Comment by mahloun** [comment]
None

**Comment by mattn** [comment]
okay, but you must get error code. try to see `echo $?` on unix, `echo %ERRORLEVEL%` on windows.

**Comment by mahloun** [comment]
Sorry I wasn't clear. It gaves me what I wanted: my username

```
$> git config --global github.user
mahloun
$> cat ~/.gitconfig
[...]
[github]
  user = mahloun
[...]
```

--
  Nieto Thomas
  Freelance Back-End Developer
  
  thms.no[at]gmail.com
  +33 6 06 69 76 21
--

On 17 April 2016 at 19:00:23, mattn (notifications@github.com) wrote:

okay, but you must get error code. try to see echo $? on unix, echo %ERRORLEVEL% on windows.

**Comment by mattn** [comment]
And do you mean that you get always error with `system("git config --get github.user")`. Right?

**Comment by mahloun** [comment]
Nope exactly the contrary. what I was wondering is why not put the version that you just proposed (which is the working one) in the script? I realised that at the 39th line still contained the following version: `s:system("git ...`

**Comment by mattn** [comment]
I don't still understand what problem you get. `s:system` is defined in line 36.

**Comment by mahloun** [comment]
On line 39: without the context `s:` the whole script is working for me and not if present. 
Here what I did:
- `-let g:github_user = substitute(s:system('git config --get github.user'), "\n", '', '')`
- `+let g:github_user = substitute(system('git config --get github.user'), "\n", '', '')`

Don't know very well scripting in git but it seems that using the _bare_ `system()` is preferable for some of us.

**Comment by mattn** [comment]
what environment do you use? linux or OSX, or Windows?

**Comment by mahloun** [comment]
OSX

**Comment by mattn** [comment]
I don't have OSX, but no errors on Linux, Windows with vim. it occur on nvim?

**Comment by mahloun** [comment]
Yep on nvim

**Comment by mattn** [comment]
Then, it is a bug of nvim.

**Comment by StaverDmitry** [comment]
Hi! Used this plugin with Ubuntu nvim and everything was great, but get the same issue with OSX now. Seems like its an OS issue

**Comment by mattn** [comment]
Once remove ~/.gist-vim and please try it again.
